llvm/include/llvm/Analysis/DXILResource.h

```diff
@@ -670,6 +670,8 @@ class DXILResourceBindingInfo {
       return CBufferSpaces;
     case dxil::ResourceClass::Sampler:
       return SamplerSpaces;
+    default:
+      llvm_unreachable("Invalid resource class");
```

Comment:
This isn't quite right. We should put the `llvm_unreachable` outside of the switch so that we can still benefit from clang's `-Wcovered-switch` warnings if `dxil::ResourceClass` were ever to change.

Reply:
Good call.